### PR TITLE
fix: cargo test for non-root users

### PR DESCRIPTION
**Pull request by mmartinv**
- **fix: add `system_info` module to `utils`**
- **fix: do not run tests that require root**
- **fix: use `system_info` mod functions to gather the current user name**

Fixes: #630

### CODE REVIEW

File: util/Cargo.toml [hunk 1]
```diff
@@ -19,3 +19,4 @@ fdo-http-wrapper = { path = "../http-wrapper", version = "0.5.0", features = ["s
 serde_yaml = "0.9"
 serde_cbor = "0.11"
 serde_json = "1"
+sysinfo = "0.30.5"
```
Comment (nullr0ute) [inline util/Cargo.toml:22]: Can this functionality made a dev dependency, or will they be used for things other than tests?

Comment (mmartinv) [reply util/Cargo.toml:22]: Currently this is used in tests only. Not sure if it will be used or not in other parts of the code in the future though.

**Comment by mmartinv** [comment]
Opened a PR to update the [sysinfo crate](https://src.fedoraproject.org/rpms/rust-sysinfo/pull-request/3)

File: util/Cargo.toml (hunk 1 above)
Comment (mmartinv) [reply util/Cargo.toml:22]: Actually is used in `integration-tests` which is not part of the unit tests, so I guess it cannot be moved to `[dev-dependencies]`

File: integration-tests/tests/common/mod.rs [hunk 2]
```diff
@@ -808,9 +810,7 @@ impl<'a> TestServerConfigurator<'a> {
                     &self.test_context.runner_path(&self.server_number),
                 );
 
-                let output = Command::new("whoami").output().unwrap();
```
Comment (7flying) [inline integration-tests/tests/common/mod.rs:811]: We originally had a dependency for getting the current user, and changed it to calling the `whoami` command.

I'm reluctant to adding back another dependency (with adds many other dependencies on cascade) when we can get this info without any external crate

Comment (mmartinv) [reply integration-tests/tests/common/mod.rs:811]: Should I remove the `systeminfo` dependency then and use `whoami` in all cases?. It doesn't seem the more "elegant" way to do this but I understand your point...

Comment (miabbott) [reply integration-tests/tests/common/mod.rs:811]: https://github.com/fdo-rs/fido-device-onboard-rs/commit/574d4e4675af3f57ba657df6ed17578e1f5305ea

:arrow_up: change from `users` crate to `whoami`

`whoami` is part of coreutils, which seems pretty safe to rely on for integration tests.

Comment (mmartinv) [reply integration-tests/tests/common/mod.rs:811]: It looks like `whoami` has been deprecated in some distributions in favor of `id` (also in coreutils package) should I use that command instead?

Comment (miabbott) [reply integration-tests/tests/common/mod.rs:811]: Sure; ‘id -un` operates the same as `whoami` according to the manpage

File: util/Cargo.toml (hunk 1 above)
Comment (mmartinv) [reply util/Cargo.toml]: I have removed sysinfo dependency as per @7flying comments

**Review by miabbott — APPROVED** [review]
Nice separation of the commits; LGTM

**Pull request merged**